alphabet/alphabet.go

```diff
@@ -0,0 +1,50 @@
+package alphabet
+
+import "fmt"
+
+type Alphabet struct {
+	symbols  []string
+	encoding map[interface{}]int
+}
+
+type AlphabetError struct {
+	message string
+}
+
+func (e *AlphabetError) Error() string {
+	return e.message
+}
+
+func NewAlphabet(symbols []string) *Alphabet {
+	encoding := make(map[interface{}]int)
+	for i, symbol := range symbols {
+		encoding[symbol] = i
+		encoding[i] = i
+	}
+	return &Alphabet{symbols, encoding}
+}
+
+func (a *Alphabet) Encode(symbol interface{}) (int, error) {
+	code, ok := a.encoding[symbol]
+	if !ok {
+		return 0, &AlphabetError{fmt.Sprintf("Symbol %v not in alphabet", symbol)}
+	}
+	return code, nil
+}
+
```

Comment:
what could replace `c` as a more interpretable variable since `code` is already being used here?